Compilation failed at Ubuntu LTS with -DMLN_WITH_QT=ON

**Describe the bug**
An error during the compilation:

```
/home/a/Projects/maplibre-native/platform/qt/src/mbgl/http_file_source.cpp:46:5: error: no matching member function for call to 'connect'
    connect(data.first, &QNetworkReply::error, this, &HTTPFileSource::Impl::onReplyFinished);
    ^~~~~~~
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:208:36: note: candidate function not viable: no overload of 'error' matching 'const char *' for 2nd argument
    static QMetaObject::Connection connect(const QObject *sender, const char *signal,
                                   ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:211:36: note: candidate function not viable: no overload of 'error' matching 'const QMetaMethod' for 2nd argument
    static QMetaObject::Connection connect(const QObject *sender, const QMetaMethod &signal,
                                   ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:463:41: note: candidate function not viable: no overload of 'error' matching 'const char *' for 2nd argument
inline QMetaObject::Connection QObject::connect(const QObject *asender, const char *asignal,
                                        ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:228:43: note: candidate template ignored: couldn't infer template argument 'Func1'
    static inline QMetaObject::Connection connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal,
                                          ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:269:13: note: candidate template ignored: couldn't infer template argument 'Func1'
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
            ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:308:13: note: candidate template ignored: couldn't infer template argument 'Func1'
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
            ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:260:13: note: candidate function template not viable: requires 3 arguments, but 4 were provided
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
            ^
/opt/Qt/5.12.0/gcc_64/include/QtCore/qobject.h:300:13: note: candidate function template not viable: requires 3 arguments, but 4 were provided
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
            ^
1 error generated.

```


**To Reproduce**

Steps to reproduce the behavior:
1. Clone the repo: git clone --depth=1 https://github.com/maplibre/maplibre-native.git
2. download 3rd party: git submodule update --init --recursive
3. configure: make _build && cd _build && cmake -DMLN_WITH_QT=ON -DCMAKE_PREFIX_PATH=/opt/Qt/5.12.0/gcc_64/lib/cmake -DCMAKE_C_COMPILER=clang-14 -DCMAKE_CXX_COMPILER=clang++-14  -DCMAKE_INSTALL_PREFIX=./_install ..
4. Compile: make -j8 install

**Expected behavior**
The compilation is successful.

**Platform information (please complete the following information):**
 - OS: Ubuntu 22.04.3 LTS
 - Platform: Qt
 - Version: the master branch


**Additional context**
gcc-12 fails at the same place.

--- a/platform/qt/src/mbgl/http_file_source.cpp
+++ b/platform/qt/src/mbgl/http_file_source.cpp
@@ -43,7 +43,7 @@ void HTTPFileSource::Impl::request(HTTPRequest* req) {
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
     connect(data.first, &QNetworkReply::errorOccurred, this, &HTTPFileSource::Impl::onReplyFinished);
 #else
-    connect(data.first, &QNetworkReply::error, this, &HTTPFileSource::Impl::onReplyFinished);
+    connect(data.first, SIGNAL(error(QNetworkReply::NetworkError)), this, SLOT(onReplyFinished()));
 #endif
 }
 

